app/policies/g5_updatable/client_policy.rb

```diff
@@ -9,9 +9,13 @@ def resolve
 
       def client_roles
         G5Authenticatable::Role
-          .joins('INNER JOIN g5_updatable_clients ON g5_updatable_clients.id = g5_authenticatable_roles.resource_id')
-          .joins('INNER JOIN g5_authenticatable_users_roles ON g5_authenticatable_roles.id = g5_authenticatable_users_roles.role_id')
-          .where('g5_authenticatable_roles.resource_type = ? and g5_authenticatable_users_roles.user_id = ?', G5Updatable::Client.name, user.id)
+          .joins('INNER JOIN g5_updatable_clients as c ON c.id = g5_authenticatable_roles.resource_id')
+          .joins('INNER JOIN g5_authenticatable_users_roles as ur ON g5_authenticatable_roles.id = ur.role_id')
+          .where('g5_authenticatable_roles.resource_type = ? and ur.user_id = ?', G5Updatable::Client.name, user.id)
+      end
+
+      def clients_from_client_and_location_roles
+        resolve | G5Updatable::LocationPolicy::Scope.new(user, G5Updatable::Location).clients_from_location_roles
       end
```

Comment:
instead of creating a new method to use, try to add it directly to the `resolve` method in this same class. 

```
      def resolve
        return scope.all if has_global_role?
        partial_scope = scope.where(id: client_roles.map(&:resource_id))
        partial_scope.empty? G5Updatable::LocationPolicy::Scope.new(user, G5Updatable::Location).clients_from_location_roles : partial_scope
      end
```